### PR TITLE
Add queue exit on timeout subroutine to the queue loop.

### PR DESCRIPTION
**Pull request by StanleyAlbayeros**
Add queue exit on timeout subroutine to the queue loop. Timeout is configurable through the YAML. 150 secs seems a good base value.

### CODE REVIEW

**Review by akshualy — CHANGES_REQUESTED** [review]
Thanks for your interest in contributing! I'm not sure this helps with the issue you've linked this PR too, but I have never ran into the described issue there so I'm willing to give this solution a shot. Just needs a few changes :)

**Comment by StanleyAlbayeros** [comment]
Heyo! It seems to check out, I have 2 bots running to check consistency issues, one with my PR and one with the latest release and the one with the release got ''hit'' with the endless queue while mine dodged it successfully. Release bot was in queue for 67 minutes before I stopped it manually.

I'll check the errors you pointed out, it seems I didn't fully commit my changes as I recognize these as stuff needed done to get it to work, including adding the exit queue button required to click it🫡

File: main.py [hunk 1]
```diff
@@ -77,17 +77,19 @@ async def wait_for_accept_button(adb_instance: ADB):
         search_result = screen.get_button_on_screen(screenshot, Button.accept)
 
 
-async def queue(adb_instance: ADB):
+async def queue(adb_instance: ADB, config: AluneConfig):
     """
     Utility method to queue a match.
 
     Args:
         adb_instance: An instance of the ADB connection to click in.
```
[the remark this answers is not in the document]
Comment (StanleyAlbayeros) [reply main.py:85]: I believe this is covered by '' async def queue(adb_instance: ADB, config: AluneConfig): '', right?

File: alune/resources/config.yaml [hunk 2]
```diff
@@ -8,6 +8,10 @@ log_level: "INFO"
 # Please note that changing this may require changing traits as well.
 game_mode: "normal"
 
+# Timeout for the bot to stay in the queue. I've seen some instances where the bot is in a queue for several dozen minutes.
+# This is the maximum time the bot will wait in the queue before it will exit the game. 150 secs seems to be a good value.
```
Comment (akshualy) [inline alune/resources/config.yaml:12]: Please leave out personal comments from the config. The following is fine (feel free to change wording, it's a suggestion):
```
# Maximum amount of time in seconds the bot should stay in queue before restarting it.
# This prevents endless queues. Increase this on servers with less player population.
```

File: main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (akshualy) [reply main.py:85]: No, I meant the line I commented on - the doc string :)

**Review by akshualy — CHANGES_REQUESTED** [review]
LGTM, just some minor wording changes I'd like to see left!

**Review by akshualy — APPROVED** [review]

**Comment by akshualy** [comment]
If you don't mind, I'll wait with the executable release for this for a few more changes I'm doing at the moment. Let me know if you'd like a separate release for this.

**Comment by StanleyAlbayeros** [comment]
> Thanks a lot! If you don't mind, I'll wait with the executable release for this for a few more changes I'm doing at the moment. Let me know if you'd like a separate release for this.

don't worry, I'm running it locally from source anyways, and this doesn't merit a single release IMO. ty!

**Pull request merged**